swarm/api/http/server.go

```diff
@@ -551,49 +551,32 @@ func (s *Server) HandlePostResource(w http.ResponseWriter, r *Request) {
 			Respond(w, r, err.Error(), http.StatusForbidden)
 			return
 		}
-	}
-
-	if updateRequest.IsNew() {
-		err = s.api.ResourceCreate(r.Context(), &updateRequest)
-		if err != nil {
-			code, err2 := s.translateResourceError(w, r, "resource creation fail", err)
-			Respond(w, r, err2.Error(), code)
-			return
-		}
-	}
-
-	if updateRequest.IsUpdate() {
 		_, err = s.api.ResourceUpdate(r.Context(), &updateRequest.SignedResourceUpdate)
 		if err != nil {
 			Respond(w, r, err.Error(), http.StatusInternalServerError)
 			return
 		}
 	}
 
-	// at this point both possible operations (create, update or both) were successful
-	// so in case it was a new resource, then create a manifest and send it over.
-
-	if updateRequest.IsNew() {
-		// we create a manifest so we can retrieve the resource with bzz:// later
-		// this manifest has a special "resource type" manifest, and its hash is the key of the mutable resource
-		// metadata chunk (rootAddr)
-		m, err := s.api.NewResourceManifest(r.Context(), updateRequest.RootAddr().Hex())
-		if err != nil {
-			Respond(w, r, fmt.Sprintf("failed to create resource manifest: %v", err), http.StatusInternalServerError)
-			return
-		}
+	// we create a manifest so we can retrieve the resource with bzz:// later
+	// this manifest has a special "resource type" manifest, and its hash is the key of the mutable resource
+	// metadata chunk (rootAddr)
+	m, err := s.api.NewResourceManifest(r.Context(), updateRequest.ViewID())
```

Comment:
Yes. This is a TODO fix. I just didn't want to complicate the first review. I need to add some flag to the JSON indicating whether or not to create the manifest. Given there is no concept of metachunk anymore, the node doesn't really know whether you are "creating" a resource or just updating it because to it, everything is now an update.

Reply:
Fixed in #839 as part of the query strings revamp